src/hooks/opengl3.rs

```diff
@@ -124,6 +128,33 @@ unsafe extern "system" fn imgui_opengl32_wglSwapBuffers_impl(dc: HDC) {
     trampoline_wglswapbuffers(dc)
 }
 
+static mut RESOLUTION_AND_RECT: Option<Mutex<([i32; 2], RECT)>> = None;
```

Comment:
I'd avoid adding global statics if possible. Global state is hard to track.

I think this instance could be added to the `ImguiRenderer` struct instead, skipping the mutex. The `reset` function could then be changed to lock `IMGUI_RENDERER`, pull this resolution data out of it, determine whether a clean needs to happen, then do a cleanup and finally `IMGUI_RENDERER.take()`. 

This flow is a bit more complicated, but it does catch a potential concurrency error where `IMGUI_RENDERER` is `None` but `cleanup` hasn't happened yet.